zip-plus/src/code/iprs.rs

```diff
@@ -0,0 +1,149 @@
+mod pntt;
+
+use crate::{
+    code::{
+        LinearCode,
+        iprs::pntt::radix8::{FieldMulByTwiddle, MBSMulByTwiddle},
+    },
+    pcs::structs::ZipTypes,
+};
+use crypto_primitives::{FromPrimitiveWithConfig, FromWithConfig};
+use itertools::Itertools;
+use num_traits::{CheckedAdd, CheckedMul};
+use std::{iter::Sum, marker::PhantomData, ops::AddAssign};
+use zinc_utils::{from_ref::FromRef, mul_by_scalar::MulByScalar};
+
+pub use pntt::radix8::params::{PnttConfigF2_16_1, Radix8PnttParams};
+
+/// Pseudo Reed-Solomon encoder over the integers. Internally uses a
+/// radix-8 NTT-style recursion with a base Vandermonde matrix sized
+/// `base_len x base_dim` (defaults to 64x32).
+#[derive(Debug, Clone)]
+pub struct IprsCode<Zt: ZipTypes, Config: pntt::radix8::params::Config> {
```

Comment:
```suggestion
pub struct IprsCode<Zt: ZipTypes, Config: Config> {
```

maybe rename to `PnttConfig`?